sparse/impl/KokkosSparse_mdf_impl.hpp

```diff
@@ -54,27 +81,30 @@ struct MDF_discarded_fill_norm {
   using static_crs_graph_type = typename crs_matrix_type::StaticCrsGraphType;
   using col_ind_type =
       typename static_crs_graph_type::entries_type::non_const_type;
-  using values_type  = typename crs_matrix_type::values_type::non_const_type;
-  using size_type    = typename crs_matrix_type::size_type;
-  using ordinal_type = typename crs_matrix_type::ordinal_type;
-  using scalar_type  = typename crs_matrix_type::value_type;
-  using KAS          = typename Kokkos::ArithTraits<scalar_type>;
+  using values_type     = typename crs_matrix_type::values_type::non_const_type;
+  using values_mag_type = KokkosSparse::Impl::mag_mirror_view_t<values_type>;
```

Comment:
If you look [here](https://github.com/kokkos/kokkos-kernels/blob/8f3574e333d0cb3bc2047fc3816cbdb3f14aab78/sparse/src/KokkosSparse_CrsMatrix.hpp#L399), you can see the definition of `crs_matrix_type::values_type`.
I think it might be simpler to do the following for values_mag_type:
```
using values_mag_type = Kokkos::View<mag_type*, Kokkos::LayoutRight, typename crs_matrix::device_type, typename crs_matrix::MemoryTraits>
```
of course you can add `::HostMirror` if you need the mirror view?

Reply:
Fair enough -- fixed that.